lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -781,6 +1079,18 @@ void ExportVerilog::prepareHWModule(Block &block,
   }
 }
 
+void ExportVerilog::prepareHWModule(hw::HWModuleOp module,
+                                    const LoweringOptions &options) {
+  // Legalization.
+  legalizeHWModule(*module.getBodyBlock(), options);
+
+  if (options.wireSpillingHeuristic != LoweringOptions::SpillNone) {
+    EmittedExpressionSizeEstimator estimator;
+    // Spill wires to prettify verilog outputs.
+    prettifyAfterLegalization(*module.getBodyBlock(), options, estimator);
+  }
+}
```

Comment:
I intentionally separated legalization and prettify phase here. As a result prepare becomes 2-pass over IR so it might cause slight performance regression.